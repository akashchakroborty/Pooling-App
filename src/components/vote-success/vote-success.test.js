import React from "react";
import { shallow } from "enzyme";
import VoteSuccess from "./vote-success.component";

describe("VoteSuccess component", () => {
  let mockHistory;
  let wrapper;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      history: mockHistory
    };

    wrapper = shallow(<VoteSuccess {...mockProps} />);
  });

  it("should render VoteSuccess component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push when button is clicked", () => {
    wrapper.find("button").simulate("click");
    expect(mockHistory.push).toHaveBeenCalled();
  });
});
