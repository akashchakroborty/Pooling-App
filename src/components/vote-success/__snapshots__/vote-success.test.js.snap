// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render VoteSuccess component 1`] = `ShallowWrapper {}`;
